make_iterator_from_pointer.hpp

```diff
@@ -0,0 +1,256 @@
+
+        wrap_pointer_to_const_iterator &operator--();
+
+        wrap_pointer_to_const_iterator operator--(int);
+
+        wrap_pointer_to_const_iterator &operator+=(difference_type);
+
+        wrap_pointer_to_const_iterator operator+(difference_type) const;
+
+        wrap_pointer_to_const_iterator &operator-=(difference_type);
+
+        wrap_pointer_to_const_iterator operator-(difference_type) const;
+
+        wrap_pointer_to_const_iterator operator[](difference_type) const;
+
+        bool operator==(const wrap_pointer_to_const_iterator &) const;
+
+    protected:
+        pointer _ptr = nullptr;
+
+        explicit wrap_pointer_to_const_iterator(pointer);
+
+        template<typename t,std::size_t N>
+        friend class ::lghazarosyan::array;
+
+    };
+
+    template<class T>
+    class wrap_pointer_to_iterator;
+
+    template<class T>
+    class wrap_pointer_to_iterator<const T*>;
+
+    template<class T>
+    class wrap_pointer_to_iterator<T*> : public wrap_pointer_to_const_iterator<T*> {
+
+    public:
+        using value_type = T;
+        using difference_type = std::ptrdiff_t;
+        using iterator_category = std::contiguous_iterator_tag;
+        using pointer = T*;
+        using reference = T&;
+        using iterator_concept = std::contiguous_iterator_tag;
+    public:
+        reference operator*() const;
+
+        pointer operator->() const;
+
+        wrap_pointer_to_iterator &operator++();
+
+        wrap_pointer_to_iterator operator++(int);
+
+        wrap_pointer_to_iterator &operator--();
+
+        wrap_pointer_to_iterator operator--(int);
+
+        wrap_pointer_to_iterator operator[](difference_type) const;
+
+        wrap_pointer_to_iterator &operator+=(difference_type);
+
+        wrap_pointer_to_iterator operator+(difference_type) const;
+
+        wrap_pointer_to_iterator &operator-=(difference_type);
+
+        wrap_pointer_to_iterator operator-(difference_type) const;
+
+    private:
+        wrap_pointer_to_iterator(const wrap_pointer_to_const_iterator<T*> &);
+
+        explicit wrap_pointer_to_iterator(pointer);
+
+        template<typename t,std::size_t N>
+        friend class ::lghazarosyan::array;
+    };
+
+
+    template<typename T>
+    typename wrap_pointer_to_iterator<T*>::reference wrap_pointer_to_iterator<T*>::operator*() const {
+        return const_cast<reference>(*wrap_pointer_to_const_iterator<T*>::_ptr);
+    }
+
+    template<typename T>
+    typename wrap_pointer_to_iterator<T*>::pointer wrap_pointer_to_iterator<T*>::operator->() const {
+        return const_cast<pointer>(std::addressof(operator*()));
+    }
+
+    template<typename T>
+    wrap_pointer_to_iterator<T*> &wrap_pointer_to_iterator<T*>::operator++() {
+        return (*this) = wrap_pointer_to_const_iterator<T*>::operator++();
+    }
+
+    template<typename T>
+    wrap_pointer_to_iterator<T*> wrap_pointer_to_iterator<T*>::operator++(int) {
+        return wrap_pointer_to_const_iterator<T*>::operator++(0);
```

Comment:
0?

Reply:
Yeas, this means I am calling the postfix version . Could be any int value.